src/transformers/models/pop2piano/modeling_pop2piano.py

```diff
@@ -0,0 +1,1429 @@
+        not be updated during training.
+        """
+        self.transformer.encoder._freeze_parameters()
+
+    def freeze_decoder(self):
+        """
+        Calling this function will disable the gradient computation for the Whisper encoder so that its parameters will
+        not be updated during training.
+        """
+        self.transformer.decoder._freeze_parameters()
+
+    def resize_token_embeddings(self, new_num_tokens: int) -> nn.Embedding:
+        new_embeddings = self.transformer.resize_token_embeddings(new_num_tokens)
+        self.config.vocab_size = new_num_tokens
+        return new_embeddings
+
+    def _freeze_parameters(self):
+        for param in self.parameters():
+            param.requires_grad = False
+        self._requires_grad = False
+
+    def forward(self,
+                inputs: BatchFeature,
+                composer="composer1",
+                max_batch_size:int = None,
+                n_bars:int = 2,
+                return_dict:Optional[bool] = False,
+                **kwargs
+                ) -> Union[Tuple[torch.Tensor], Seq2SeqLMOutput] :
+
+        # select composer randomly if not already given
+        composer_to_feature_token = self.config.composer_to_feature_token
+        if composer is None:
+            composer = np.random.choice(list(composer_to_feature_token.keys()), size=1)[0]
+        elif composer not in composer_to_feature_token.keys():
+            raise ValueError(f"Composer not found in list, Please choose from {list(composer_to_feature_token.keys())}")
+
+        n_bars = self.config.dataset.get("n_bars", None) if n_bars is None else n_bars
+        max_batch_size = 64 // n_bars if max_batch_size is None else max_batch_size
+        max_length = self.config.dataset.get("target_length") * max(1, (n_bars // self.config.dataset.get("n_bars")))
+
+        relative_tokens_arr = []
+        # For multiple audios(batch, seq_len, seq_dim)
+        for i, input_embeds in enumerate(inputs["input_features"]):
+            input_embeds = self.spectrogram(input_embeds).transpose(-1, -2)
+            if self.config.dataset.get("mel_is_conditioned", None):
+                composer_value = composer_to_feature_token[composer]
+                composer_value = torch.tensor(composer_value, device=self.device)
+                composer_value = composer_value.repeat(input_embeds.shape[0])
+                input_embeds = self.mel_conditioner(input_embeds, composer_value)
+
+            # As described in the official pop2piano implementation
+            relative_tokens = self.transformer.generate(inputs_embeds=input_embeds,
+                                                        max_length=max_length,
+                                                        **kwargs
+                                                        )
+            relative_tokens_arr.append(relative_tokens)
+
+        if return_dict:
+            return BackboneOutput(feature_maps=relative_tokens_arr)
```

Comment:
Since different audio outputs have different shapes (`12, 123`, `120, 100`) it would not be possible to pad them that's why I used  `BackboneOutput`, since then I can put them in a `list`,